model_api/cpp/adapters/src/openvino_adapter.cpp

```diff
@@ -17,15 +17,25 @@
 #include "adapters/openvino_adapter.h"
 #include <openvino/openvino.hpp>
 #include <utils/slog.hpp>
+#include <stdexcept>
 #include <vector>
 
 void OpenVINOInferenceAdapter::loadModel(const std::shared_ptr<const ov::Model>& model, ov::Core& core,
-                                                            const std::string& device, const ov::AnyMap& compilationConfig) {
+                                                            const std::string& device, const ov::AnyMap& compilationConfig, size_t max_num_requests) {
     slog::info << "Loading model to the plugin" << slog::endl;
-
-    compiledModel = core.compile_model(model, device, compilationConfig);
-    inferRequest = compiledModel.create_infer_request();
-
+    ov::AnyMap customCompilationConfig(compilationConfig);
+    if (max_num_requests != 1) {
+        if (customCompilationConfig.find("PERFORMANCE_HINT") == customCompilationConfig.end()) {
+            customCompilationConfig["PERFORMANCE_HINT"] = ov::hint::PerformanceMode::THROUGHPUT;
```

Comment:
> Maybe it will even switch latency to throughput automatically.

I haven't found that in the docs, so I experimented a bit.
The optimal number of requests is not known in advance, so basically we have several options:
1) user sets the number explicitly, throughput mode is enabled explicitly if we have more that 1 request
2) user relies on `model.get_property(ov::optimal_number_of_infer_requests)`, and creates that many requests manually:
if throughput mode is not set, then no speedup
3) ov::hint::num_requests is passed, throughput mode is not set:
`model.get_property(ov::optimal_number_of_infer_requests)` is 1. My assumption this is because throughput mode is not set under the hood.
4) Like in docs, `ov::hint::performance_mode(ov::hint::PerformanceMode::THROUGHPUT),
    ov::hint::num_requests(4)` are set. 
`model.get_property(ov::optimal_number_of_infer_requests)` is 4, exactly as suggested in the hint

Among these option I'd prefer 1 and 2 which are already implemented.
Besides, to support `ov::hint`, we need to get rid of AnyMap compilation configuration.

Reply:
What's wrong with (4)? I believe that should be preferred. My guess is that `model.get_property(ov::optimal_number_of_infer_requests)` is 4 because `num_requests(4)` sets throughput to use 4 streams. If you don't provide `num_requests`, the number of streams is equal to the number of threads for `CPU`. Using 4 ireqs only leaves some of the streams unused.